eng/illink.targets

```diff
@@ -64,6 +75,36 @@
     </BinPlaceTargetFramework>
   </ItemGroup>
 
+  <Target Name="_EmbedILLinkSubstitutionsXmls"
+          Condition="'$(ILLinkTrimAssembly)' == 'true'"
+          DependsOnTargets="_CombineILLinkSubstitutionsXmls">
+
+    <ItemGroup Condition="'$(ILLinkSubstitutionsXml)' != ''">
+      <EmbeddedResource Include="$(ILLinkSubstitutionsXml)">
+        <LogicalName>ILLink.Substitutions.xml</LogicalName>
+      </EmbeddedResource>
+    </ItemGroup>
+
+  </Target>
+
+  <UsingTask TaskName="CombineLinkerXmlFiles" AssemblyFile="$(ILLinkTasksPath)" />
+  <Target Name="_CombineILLinkSubstitutionsXmls"
+          Condition="'$(ILLinkSubstitutionsXml)' == '' and '@(ILLinkSubstitutionsXml)' != ''"
+          Inputs="@(ILLinkSubstitutionsXml)"
+          Outputs="$(ILLinkSubstitutionsXmlIntermediatePath)">
+    <PropertyGroup>
+      <ILLinkSubstitutionsXml>$(ILLinkSubstitutionsXmlIntermediatePath)</ILLinkSubstitutionsXml>
```

Comment:
NIT: do you need this extra property or can you just use the same one (ILLinkSubstitutionsXmlIntermediatePath)

Reply:
Setting `$(ILLinkSubstitutionsXml)` is how you get the `ILLink.Substitutions.xml` file embedded into the assembly. See:

```xml
    <ItemGroup Condition="'$(ILLinkSubstitutionsXml)' != ''">
      <EmbeddedResource Include="$(ILLinkSubstitutionsXml)">
        <LogicalName>ILLink.Substitutions.xml</LogicalName>
      </EmbeddedResource>
    </ItemGroup>
```

So what this is doing is setting that property so the `_EmbedILLinkSubstitutionsXmls` actually embeds the file. The separate `$(ILLinkSubstitutionsXmlIntermediatePath)` property is necessary in order to get the `Inputs/Outputs` working correctly so it is incremental.

Reply:
Got it 👍